staging/kos/pkg/registry/network/subnet/strategy.go

```diff
@@ -89,49 +89,20 @@ func (*subnetStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object)
 }
 
 func (*subnetStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
-}
-
-func (s *subnetStrategy) Validate(ctx context.Context, obj runtime.Object) field.ErrorList {
-	var errs field.ErrorList
-
-	thisSubnet := obj.(*network.Subnet)
-
-	vniRangeErrs := s.checkVNIRange(thisSubnet)
-	errs = append(errs, vniRangeErrs...)
-
-	cidrFormatErrs := s.checkCIDRFormat(thisSubnet)
-	errs = append(errs, cidrFormatErrs...)
-
-	if len(vniRangeErrs) > 0 {
-		// If we are here, the VNI range of the created subnet is invalid. All
-		// of the next validation steps make sense so long as the vni is valid,
-		// thus we return immediately.
-		return errs
-	}
-
-	// We need to check the under-validation subnet against subnets with the
```

Comment:
I think it makes sense to retain this pre-emptive checking in the apiservers.  While it does not prevent conflicts in 100% of the cases, it will in most cases --- and earlier detection of a problem is better than later detection.

Reply:
When I removed these checks I was reasoning as a "purist": conflicts should be handled uniformly. Having both the validator and the pre-emptive checks violates that: in some cases creation of conflicting subnets fails, while in others it succeeds albeit subnets are later labelled as not valid. But on further reflection I agree with reintroducing pre-emptive checks. The value for a user is much higher with them, and even with the validator alone you don't have truly uniform behavior (in case of conflicting subnets sometimes both are labelled as not validated while other times only one is).

Reply:
so you are going to put those checks back?

Reply:
Yes, but if you agree I would make the pre-emptive checks conditional on an api-server flag enabled by default. The rationale is that I want to be able to test the subnets validator, but as you said the pre-emptive checks would catch most conflicts before the validator can see them.

Reply:
Yes, that's a good idea.

Reply:
Done in 7304a04.